### PR TITLE
feat: add list of available models in the sidenav

### LINKED ISSUE #29: List of items in sidenav
As a user
When I click on the hamburger menu of the headbar
Then I should see a list of available entities in the sidenav

### PR DESCRIPTION
**Pull request by Gregcop1**
Fix #29 

Changes proposed in this pull request:
- [x] header bar with hamburger menu
- [x] list of models in the sidebar
- [x] route for home (empty component)
- [x] route for list (empty component)

@Simperfit 

Make sure those checkboxes are checked if you want to be reviewed
- [x] Your branch should be mergeable on current master
- [x] Tests should pass
- [x] Linter should pass
- [x] Code coverage should stay higher than 90%
- [x] Squash your commits to one commit and follow those recommendations about [writing good commit message](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#-git-commit-guidelines)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +56,7 @@
     "zone.js": "^0.6.12"
   },
   "devDependencies": {
+    "@angular2-material/list": "^2.0.0-alpha.5",
```
Comment (toofff) [inline package.json]: `angular2-material/list` is a development dependency?

File: package.json [hunk 2]
```diff
@@ -42,6 +42,10 @@
     "@angular/platform-browser": "2.0.0-rc.1",
     "@angular/platform-browser-dynamic": "2.0.0-rc.1",
     "@angular/router": "2.0.0-rc.1",
+    "@angular2-material/core": "^2.0.0-alpha.4",
+    "@angular2-material/sidenav": "^2.0.0-alpha.4",
+    "@angular2-material/button": "^2.0.0-alpha.4",
+    "@angular2-material/icon": "^2.0.0-alpha.4",
```
Comment (gorghoa) [inline package.json:48]: should’nt we upgrade to alpha.5 ? out of PR scope maybe

File: src/app/hm-admin/header/header.component.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Component, OnInit } from '@angular/core';
+import {MdButton} from '@angular2-material/button';
+import {MdIcon, MdIconRegistry} from '@angular2-material/icon';
```
Comment (gorghoa) [inline src/app/hm-admin/header/header.component.ts]: MdIconRegistry seems unused

File: src/app/hm-admin/hm-admin.component.ts [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
+import {Route} from '@angular/router';
+import {MdButton} from '@angular2-material/button';
+import {MdIcon, MdIconRegistry} from '@angular2-material/icon';
 import {ConfigService} from './services/config/Config.service';
 import {CoreDefinitionService} from './services/core/CoreDefinition.service';
 import {SchemaService} from './services/schema/schema.service';
+import {HeaderComponent} from './header/index'
+import {ContentComponent, SidenavService} from './content/index'
+import {HomeComponent} from './+home';
+import { ListComponent } from './+list';
```
Comment (gorghoa) [inline src/app/hm-admin/hm-admin.component.ts]: are we spaces nazis in this project? If so, well, there is two spaces here that we should drop

File: src/app/hm-admin/hm-admin.component.ts [hunk 5]
```diff
@@ -1,23 +1,46 @@
 import {Component, Input, OnInit} from '@angular/core';
 import {HTTP_PROVIDERS} from '@angular/http';
+import {Route} from '@angular/router';
+import {MdButton} from '@angular2-material/button';
+import {MdIcon, MdIconRegistry} from '@angular2-material/icon';
 import {ConfigService} from './services/config/Config.service';
 import {CoreDefinitionService} from './services/core/CoreDefinition.service';
 import {SchemaService} from './services/schema/schema.service';
+import {HeaderComponent} from './header/index'
```
Comment (toofff) [inline src/app/hm-admin/hm-admin.component.ts]: forget the semicolon?

File: src/app/hm-admin/hm-admin.component.ts [hunk 6]
```diff
@@ -1,23 +1,46 @@
...
+import {ContentComponent, SidenavService} from './content/index'
```
Comment (toofff) [inline src/app/hm-admin/hm-admin.component.ts]: forget the semicolon?

File: package.json (hunk 1 above)
Comment (Simperfit) [inline package.json]: and alpha.5 ? the other are in .4

File: src/app/hm-admin/hm-admin.component.ts [hunk 7]
```diff
@@ -1,23 +1,46 @@
...
+import {HomeComponent} from './+home';
+import { ListComponent } from './+list';
+
+export const HmAdminRoutes: Route[] = [
+  // @todo remove after useAsDefault or redirectTo implementation
+  new Route({path: '/', component: HomeComponent}),
+  new Route({path: '/home', component: HomeComponent}),
```
Comment (Simperfit) [inline src/app/hm-admin/hm-admin.component.ts:17]: Do we really need this ?

File: src/app/hm-admin/hm-admin.component.ts [hunk 8]
```diff
@@ -1,23 +1,46 @@
...
+  directives: [MdButton, MdIcon, HeaderComponent, ContentComponent]
 })
 export class HmAdminComponent implements OnInit {
   @Input()
   config: any;
 
   constructor(private configService: ConfigService,
               private coreDefinitionService: CoreDefinitionService,
-              private schemaService: SchemaService) {}
+              private schemaService: SchemaService) {
```
Comment (Simperfit) [inline src/app/hm-admin/hm-admin.component.ts]: why this change ?

File: src/index.html [hunk 9]
```diff
@@ -1,12 +1,13 @@
 <!DOCTYPE html>
 <html>
 <head>
-    <meta charset="utf-8">
-    <title>Tst</title>
-    <base href="/">
-    {{content-for 'head'}}
-    <link rel="icon" type="image/x-icon" href="favicon.ico">
-    <meta name="viewport" content="width=device-width, initial-scale=1">
+  <meta charset="utf-8">
```
Comment (Simperfit) [inline src/index.html:4]: why no more space ?

File: package.json (hunk 2 above)
Comment (Gregcop1) [inline package.json:48]: the ^ allow npm to upgrade automatically. BTW, it seems we have to use 5-2 cause the previous build was pretty crappy.

File: src/app/hm-admin/hm-admin.component.ts (hunk 7 above)
Comment (Gregcop1) [inline src/app/hm-admin/hm-admin.component.ts:17]: What ? the Home component ? Sure, it could be replaced by a custom dashboard and by a login page when we'll have one.

File: src/app/hm-admin/hm-admin.component.ts (hunk 8 above)
Comment (Gregcop1) [inline src/app/hm-admin/hm-admin.component.ts]: auto-align. It's fixed

File: src/index.html (hunk 9 above)
Comment (Gregcop1) [inline src/index.html:4]: According to the .editorconfig html files have to spaces for indent, not 4

File: src/app/hm-admin/hm-admin.component.ts (hunk 7 above)
Comment (Simperfit) [inline src/app/hm-admin/hm-admin.component.ts:17]: I was not talking about the component but about the route, why one more route ?

Comment (Simperfit) [inline src/app/hm-admin/hm-admin.component.ts:17]: But you have to rewrite the route for the 3.0 router anyway.

**Comment by Simperfit** [comment]
You have my +1 after the comments :)

File: src/app/hm-admin/hm-admin.component.ts (hunk 7 above)
Comment (Gregcop1) [inline src/app/hm-admin/hm-admin.component.ts:17]: Let the router take some age before switching... just to be sure ;)
